test/unit/deadline_client/ui/test_host_requirements_dataclass_serialize.py

```diff
@@ -0,0 +1,178 @@
+
+
+def test_host_requirements_as_objects_serialize():
+    requirements = HostRequirements(
+        os_requirements=OsRequirements(
+            operating_systems=[OsRequirements.WINDOWS],  # type: ignore
+            cpus=[OsRequirements.X86_64],  # type: ignore
+        ),
+        hardware_requirements=HardwareRequirements(
+            cpu_min=8,
+            cpu_max=64,
+            memory_min=16 * 1024,
+            memory_max=128 * 1024,
+        ),
+        custom_requirements=CustomRequirements(
+            amounts=[
+                CustomAmountRequirement(name="Bugs", min=1, max=10),
+            ],
+            attributes=[
+                CustomAttributeRequirement(
+                    name="pipelineFeatures",
+                    option=CustomAttributeRequirement.ANY_OF,
+                    values=["feature1", "feature2"],
+                ),
+            ],
+        ),
+    )
+    open_jd_requirements = requirements.serialize()
+    _compare_requirements(open_jd_requirements, HOST_REQUIREMENTS_1)
+
+
+def test_host_requirements_as_dictionary_serialize():
+    requirements = HostRequirements(
+        os_requirements={  # type: ignore
+            "operating_systems": ["windows"],
+            "cpus": ["x86_64"],
+        },
+        hardware_requirements={  # type: ignore
+            "cpu_min": 8,
+            "cpu_max": 64,
+            "memory_min": 16 * 1024,
+            "memory_max": 128 * 1024,
+        },
+        custom_requirements={  # type: ignore
+            "amounts": [
+                {"name": "Bugs", "min": 1, "max": 10},
+            ],
+            "attributes": [
+                {"name": "pipelineFeatures", "option": "anyOf", "values": ["feature1", "feature2"]},
+            ],
+        },
+    )
+    open_jd_requirements = requirements.serialize()
+    _compare_requirements(open_jd_requirements, HOST_REQUIREMENTS_1)
+
+
+def test_host_requirements_invalid_os():
+    try:
+        requirements = HostRequirements(
+            os_requirements=OsRequirements(
+                operating_systems=["freebsd"],  # type: ignore
+                cpus=[OsRequirements.X86_64],  # type: ignore
+            ),
+            hardware_requirements=HardwareRequirements(
+                cpu_min=8,
+                cpu_max=64,
+                memory_min=16 * 1024,
+                memory_max=128 * 1024,
+            ),
+            custom_requirements=CustomRequirements(
+                amounts=[
+                    CustomAmountRequirement(name="Bugs", min=1, max=10),
+                ],
+                attributes=[
+                    CustomAttributeRequirement(
+                        name="pipelineFeatures",
+                        option=CustomAttributeRequirement.ANY_OF,
+                        values=["feature1", "feature2"],
+                    ),
+                ],
+            ),
+        )
+        requirements.serialize()
+    except ValueError:
```

Comment:
Instead of this try-except, does `pytest.raises(ValueError):` work? It's a little bit simpler and has better messaging if it fails.

Similar note for the next test.

Reply:
Good catch, I am less familiar with pytest so I wasn't aware of that